android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java

```diff
@@ -483,6 +483,7 @@ public void restartScan() {
       urlDeviceDiscoverer.stopScan();
     }
     mScanStartTime = new Date().getTime();
+    mPwCollection = new PhysicalWebCollection();
```

Comment:
This clears the cached data. Otherwise when you manually refresh you still get the cached data along with scanning for new beacons.